prov/efa/src/efa_cq.c

```diff
@@ -118,16 +117,39 @@ static void efa_cq_read_data_entry(struct efa_wc *wc, int i, void *buf)
 	entry[i].len = (uint64_t)wc->ibv_wc.byte_len;
 }
 
+/* Unlike ibv_poll_cq, wide completion APIs do not write to ibv_wc struct. We need to do this manually.
+ * This is inspired by rdma-core, i.e. `efa_process_cqe` and `efa_process_ex_cqe`.
+ */
+static inline void efa_unsafe_write_ibv_cq_ex_wc(struct ibv_cq_ex *cq, struct ibv_wc *wc) {
```

Comment:
The extended CQ verbs have been available in rdma-core efa provider since since Aug. 2020. We can assume that they will work in recent versions.

Reply:
what is the version of that rdma-core?

Reply:
Function name does not look right, should be something like `efa_cq_cw_from_ibv_cq_unsafe`.

Reply:
```
[ec2-user@ip-10-0-15-237 rdma-core]$ git tag --contains 40ee2e2a7df82e5ed1a58fb843474795ed52ffeb
debian/31.0-1
debian/32.0-1
debian/33.0-1
debian/33.1+git20210317-1
debian/33.1-1
debian/33.2-1
debian/36.0-1
debian/36.0-2
debian/38.0-1
debian/39.0-1
v31.0
v31.1
v31.2
v31.3
v31.4
v31.5
v31.7
v32.0
v32.1
v32.2
v32.3
v32.5
v32.6
v33.0
v33.1
v33.2
v33.3
v33.5
v33.6
v34.0
v34.1
v34.2
v34.4
v34.5
v35.0
v35.1
v35.3
v35.4
v36.0
v36.1
v36.3
v36.4
v37.0
v37.1
v37.2
v37.3
v38.0
v38.1
v38.2
v39.0
v39.1
v40.0
```

Reply:
> Function name does not look right, should be something like `efa_cq_cw_from_ibv_cq_unsafe`.

Sure. I will change the function name.

Reply:
That is not a very old version. It does not hurt to check some of the API's availability in https://github.com/ofiwg/libfabric/blob/main/prov/efa/configure.m4#L44-L57

Reply:
> That is not a very old version. It does not hurt to check some of the API's availability in https://github.com/ofiwg/libfabric/blob/main/prov/efa/configure.m4#L44-L57

Chatted with @shijin-aws offline. rdma-core versions older than 31.0 should not be used anymore. We don't need to check the extended CQ verbs.